File: 55.cpp
struct Solution {
    inline bool canJump(vector<int>& nums) {
        int n = nums.size();
        int m = n - 1;
        for (int i = n - 2; i >= 0; --i){
            if (i + nums[i] >= m){
                m = i;
            }
        }
        return m == 0;
    }
};

File: sol/55.py

class Solution:
    def canJump(self, nums: List[int]) -> bool:
        n = len(nums)
        minReach = n - 1
        for i in range(n - 2, -1, -1):
            if i + nums[i] >= minReach:
                minReach = i
        return minReach == 0
